optuna/samplers/_nsga2.py

```diff
@@ -327,3 +393,52 @@ def _crowding_distance_sort(population: List[FrozenTrial]) -> None:
 
     population.sort(key=lambda x: manhattan_distances[x.number])
     population.reverse()
+
+
+def _constrained_dominates(
+    trial0: FrozenTrial, trial1: FrozenTrial, directions: Sequence[StudyDirection]
+) -> bool:
+    """Checks constrained-domination.
+
+    A trial x is said to constrained-dominate a trial y, if any of the following conditions is
+    true:
+    1) Trial x is feasible and trial y is not.
+    2) Trial x and y are both infeasible, but solution x has a smaller overall constraint
+    violation.
+    3) Trial x and y are feasible and trial x dominates trial y.
+    """
+
+    constraints0 = trial0.system_attrs[_CONSTRAINTS_KEY]
+    constraints1 = trial1.system_attrs[_CONSTRAINTS_KEY]
+
+    assert isinstance(constraints0, (list, tuple))
+    assert isinstance(constraints1, (list, tuple))
+
+    if len(constraints1) != len(constraints1):
```

Comment:
Ops, seems like a typo.
```suggestion
    if len(constraints0) != len(constraints1):
```

Reply:
Thanks! I fixed it.